src/main/java/xyz/kyngs/herbot/handlers/command/AbstractCommand.java

```diff
@@ -1,17 +1,114 @@
 package xyz.kyngs.herbot.handlers.command;
 
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
 import xyz.kyngs.herbot.HerBot;
+import xyz.kyngs.herbot.handlers.command.argument.AbstractArgument;
+import xyz.kyngs.herbot.handlers.command.argument.Arguments;
+import xyz.kyngs.herbot.handlers.user.UserProfile;
+import xyz.kyngs.herbot.util.ExceptionUtil;
+import xyz.kyngs.herbot.util.ImmutableEntry;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
 
 public abstract class AbstractCommand implements CommandExecutor {
 
     protected final HerBot herBot;
     protected final String description;
+    private final List<AbstractArgument> args;
+
+    public String getPermission() {
+        return permission;
+    }
+
+    private final String permission;
 
-    public AbstractCommand(HerBot herBot, String description) {
+    public AbstractCommand(HerBot herBot, String description, String permission) {
         this.herBot = herBot;
         this.description = description;
+        this.permission = permission;
+        args = new ArrayList<>();
     }
 
+    public void addArg(AbstractArgument arg) {
+        args.add(arg);
+    }
+
+    @Override
+    public void onCommand(User author, Guild guild, TextChannel channel, Message message, String[] args, UserProfile profile, GuildMessageReceivedEvent event) {
+        if (!checkPerm(message, profile) || !checkArgSize(message, args) || !checkArgumentType(message, args)) return;
```

Comment:
You could remove negatiions by switching the return values from the functions, but the name of the function might be misleading afterwards.

Reply:
No I don't think so, all the functions return true if the thing that the function checks is ok, on this line i check whether something is not ok (using negation) and stop the code from executing.